target/linux/ath79/dts/qca9531_comfast_cf-e5.dts

```diff
@@ -0,0 +1,170 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca953x.dtsi"
+
+/ {
+	compatible = "comfast,cf-e5", "qca,qca9531";
+	model = "COMFAST CF-E5/E7";
+
+	keys {
+		compatible = "gpio-keys-polled";
+
+		poll-interval = <20>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&jtag_disable_pins &led_wan_pin>;
+
+		button0 {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		wan {
+			label = "cf-e5:blue:wan";
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+		};
+
+		lan {
+			label = "cf-e5:blue:lan";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan {
+			label = "cf-e5:blue:wlan";
+			gpios = <&gpio 0 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		rssi0 {
+			label = "cf-e5:blue:rssi0";
+			gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+		};
+
+		rssi1 {
+			label = "cf-e5:blue:rssi1";
+			gpios = <&gpio 3 GPIO_ACTIVE_LOW>;
+		};
+
+		rssi2 {
+			label = "cf-e5:blue:rssi2";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	gpio-export {
```

Comment:
@fall513 We're trying to get rid of the `gpio-export` functionality as it's hack for missing kernel functionality, which was rejected in upstream kernel long time ago, for details see this email http://lists.infradead.org/pipermail/openwrt-devel/2019-February/015772.html, discussion in #1366 or https://github.com/openwrt/openwrt/pull/1814#issuecomment-462942022.

So I'm wondering if it would be possible to replace the `gpio-export` construct with the userspace gpio-switch alternative, for example https://github.com/openwrt/openwrt/blob/master/target/linux/layerscape/base-files/etc/board.d/03_gpio_switches#L20-L22

Reply:
OK, I will modify and do some test.